framework/wazuh/core/agent.py

```diff
@@ -976,28 +976,30 @@ def add_group_to_agent(group_id, agent_id, force=False, replace=False, replace_l
         :param replace_list: List of Group names that can be replaced
         :return: Agent ID.
         """
+        agent = Agent(agent_id)
+        agent_info = False
         if replace_list is None:
             replace_list = []
         if not force:
             # Check if agent exists, it is not 000 and the group exists
-            Agent(agent_id).get_basic_information()
-
             if agent_id == "000":
                 raise WazuhError(1703)
 
             if not Agent.group_exists(group_id):
                 raise WazuhResourceNotFound(1710)
 
+            agent.load_info_from_db()
+            agent_info = True
+
         # Get agent's group
         group_path = path.join(common.groups_path, agent_id)
         try:
             with open(group_path) as f:
-                multigroup_name = f.read().replace('\n', '')
+                multigroup_name = f.read().strip()
         except Exception as e:
             # Check if agent is never_connected.
-            failed = Agent(agent_id)
-            failed.load_info_from_db()
-            if failed.status == 'never_connected':
+            not agent_info and agent.load_info_from_db()
```

Comment:
```suggestion
            agent.load_info_from_db()
```

Reply:
The above lines do not add anything to the code. We can reduce the size of this function by eliminating them and always getting the agent information.

Reply:
The above lines do not add anything to the code. We can reduce the size of this function by eliminating them and always getting the agent information.